File: leet_easy_python/leet_101/leet_101b.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def isSymmetric(self, root):
        if not root: return True
        left = [root.left]
        right = [root.right]
        while left and right:
            l = left.pop()
            r = right.pop()
            if not l and not r: pass #do nothing         
            elif not l or not r: return False
            elif l.val != r.val: return False
            else:
                left.append(l.left)
                left.append(l.right)
                right.append(r.right)
                right.append(r.left)
        return True
        
        """
        :type root: TreeNode
        :rtype: bool
        """
        

File: leet_easy_java/leet_101/leet_101b.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public boolean isSymmetric(TreeNode root) {
        if(root == null) return true;
        Stack<TreeNode> stk = new Stack<TreeNode>();
        stk.push(root.left);
        stk.push(root.right);
        while(!stk.empty()){
            TreeNode a = stk.pop();
            TreeNode b = stk.pop();
            if(a == null && b == null){
                //~ do nothing
            }else if (a == null || b == null){
                return false;
            }else if(a.val != b.val){
                return false;
            }else{
                stk.push(a.left);
                stk.push(b.right);
                stk.push(a.right);
                stk.push(b.left);
            }
        }
        return true;
    }
}